deployment/ccip/sequence/evm/v1_6/seq_deploy_chain.go

```diff
@@ -0,0 +1,362 @@
+package v1_6
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	"github.com/ethereum/go-ethereum/common"
+	"golang.org/x/sync/errgroup"
+
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/v1_6_0/fee_quoter"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/v1_6_0/nonce_manager"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/v1_6_0/offramp"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/v1_6_0/onramp"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/v1_6_0/rmn_home"
+	"github.com/smartcontractkit/chainlink-ccip/chains/evm/gobindings/generated/v1_6_0/rmn_remote"
+	"github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+	"github.com/smartcontractkit/chainlink-deployments-framework/operations"
+
+	cldf "github.com/smartcontractkit/chainlink-deployments-framework/deployment"
+
+	ccipopsv1_2 "github.com/smartcontractkit/chainlink/deployment/ccip/operations/evm/v1_2"
+	ccipopsv1_6 "github.com/smartcontractkit/chainlink/deployment/ccip/operations/evm/v1_6"
+	"github.com/smartcontractkit/chainlink/deployment/ccip/shared/opsutil"
+)
+
+type DeployChainContractsConfig struct {
+	HomeChainSelector      uint64
+	ContractParamsPerChain map[uint64]ChainContractParams
+}
+
+func (c DeployChainContractsConfig) Validate() error {
+	if err := deployment.IsValidChainSelector(c.HomeChainSelector); err != nil {
+		return fmt.Errorf("invalid home chain selector: %d - %w", c.HomeChainSelector, err)
+	}
+	for cs, args := range c.ContractParamsPerChain {
+		if err := deployment.IsValidChainSelector(cs); err != nil {
+			return fmt.Errorf("invalid chain selector: %d - %w", cs, err)
+		}
+		if err := args.Validate(); err != nil {
+			return fmt.Errorf("invalid contract args for chain %d: %w", cs, err)
+		}
+	}
+	return nil
+}
+
+type ChainContractParams struct {
+	FeeQuoterParams ccipopsv1_6.FeeQuoterParams
+	OffRampParams   ccipopsv1_6.OffRampParams
+}
+
+func (c ChainContractParams) Validate() error {
+	if err := c.FeeQuoterParams.Validate(); err != nil {
+		return fmt.Errorf("invalid FeeQuoterParams: %w", err)
+	}
+	if err := c.OffRampParams.Validate(false); err != nil {
+		return fmt.Errorf("invalid OffRampParams: %w", err)
+	}
+	return nil
+}
+
+var (
```

Comment:
We could unnest this code. There's no reason to use the group syntax.

var DeployChainContractsSeq = operations.NewSequence(
    ...
)